components/terrain/material.cpp

```diff
@@ -12,66 +12,150 @@
 
 #include <components/shader/shadermanager.hpp>
 
+#include <mutex>
 
-namespace Terrain
+namespace
 {
+    class BlendmapTexMat
+    {
+    public:
+        static const osg::ref_ptr<osg::TexMat>& value(const int blendmapScale)
```

Comment:
You should probably document these methods with static variables, saying something along the lines of "not thread safe."